joplin/pages/official_documents_page/factories.py

```diff
@@ -1,15 +1,164 @@
+    for index in range(len(page_dictionaries['en']['official_documents']['edges'])):
+        en_node = page_dictionaries['en']['official_documents']['edges'][index]['node']
+        es_node = page_dictionaries['es']['official_documents']['edges'][index]['node']
+        en_filename = en_node['document']['filename']
+        es_filename = es_node['document']['filename']
+
+        combined_node = en_node
+        combined_node['title_es'] = es_node['title']
+        combined_node['authoring_office_es'] = es_node['authoring_office']
+        combined_node['summary_es'] = es_node['summary']
+        combined_node['name_es'] = es_node['name']
+        combined_node['document'] = create_document_from_importer_dictionary(en_node['document'])
+        # the api gives us the english doc for janis links if we don't have a spanish doc
+        # we're assuming we have different filenames for english and spanish docs
+        # so we only import the spanish doc if we have a different filename
+        if es_filename != en_filename:
+            combined_node['document_es'] = create_document_from_importer_dictionary(es_node['document'])
+
+        official_documents_page_documents.append(combined_node)
+    combined_dictionary['add_official_documents_page_documents'] = {'official_documents_page_documents': official_documents_page_documents}
+
+    # remove official_documents if we have it because:
+    # * we just added it up above
+    # * it has been renamed
+    # todo: why isn't pop working?
+    if 'official_documents' in combined_dictionary:
+        del combined_dictionary['official_documents']
+
+    page = OfficialDocumentPageFactory.create(**combined_dictionary)
+    return page
+
+
+def create_document_from_importer_dictionary(document_dictionary):
+    # right now we're just going off filename, so first let's see if we can download the file
+    # let's try to get it from both the staging and prod s3 buckets (since that's what janis does)
+    file_name = document_dictionary['filename']
+
+    for url in [
+        f'https://joplin-austin-gov-static.s3.amazonaws.com/production/media/documents/{file_name}',
+        f'https://joplin-austin-gov-static.s3.amazonaws.com/staging/media/documents/{file_name}'
+    ]:
+        response = requests.get(url)
+        if response.status_code == 200:
+            break
+
+    # wagtail calculates document hashes this way
+    # https://github.com/wagtail/wagtail/blob/081705fc7a2d9aec75da25a3593b490f3c145d2b/wagtail/documents/models.py#L115
+    file_hash = hashlib.sha1(response.content).hexdigest()
+
+    # Check if a document with the same hash has already been imported
+    try:
+        document = Document.objects.get(file_hash=file_hash)
+    except Document.DoesNotExist:
+        document = None
+    if document:
+        return document
+
+    # It has not been imported, let's do it!
+    document = DocumentFactory.create(file=ContentFile(response.content, name=file_name), title=file_name)
```

Comment:
I appreciate how long it took to make this work.

Reply:
FYI wagtail-factories has a document factory:
https://github.com/mvantellingen/wagtail-factories/blob/819666151f8c83317bdd0019074173ba2ade85eb/src/wagtail_factories/factories.py#L142

Which leverages factory boy's Django FileField:
https://factoryboy.readthedocs.io/en/latest/orms.html#extra-fields